nixos/modules/services/web-apps/funkwhale/funkwhale.nix

```diff
@@ -0,0 +1,329 @@
+        }
+      '';
+      virtualHosts = {
+        "${cfg.hostname}" = {
+        # enableACME = true; #Ask Let's Encrypt to sign a certificate for this vhost
+        # addSSL = true;
+        # forceSSL = true;
+          root = "${pkgs.funkwhale}/front";
+          default = true;
+          locations = {
+            "/" = {
+              proxyWebsockets = true;
+              tryFiles = "$uri $uri/ @rewrites";
+            };
+            "@rewrites".extraConfig = ''
+                rewrite ^(.+)$ /index.html last;
+              '';
+            "/api/".proxyPass = "http://funkwhale-api/api/";
+            "/federation/".proxyPass = "http://funkwhale-api/federation/";
+            "/rest/".proxyPass = "http://funkwhale-api/api/subsonic/rest/";
+            "/.well-known/".proxyPass = "http://funkwhale-api/.well-known/";
+            "/media/".alias = "${cfg.api.media_root}/";
+            "/_protected/media" = {
+              extraConfig = ''
+                internal;
+              '';
+              alias = "${cfg.api.media_root}";
+            };
+            "/_protected/music" = {
+              extraConfig = ''
+                internal;
+              '';
+              alias = "${cfg.music_directory_path}";
+            };
+            "/staticfiles/".alias = "${cfg.api.static_root}/";
+          };
+        };
+        };
+      };
+
+      systemd.services = 
+        let serviceConfig = {
+            User = "funkwhale";
+            WorkingDirectory = "${pkgs.funkwhale}";
+            EnvironmentFile =  "${funkwhaleEnvFile}";
+            # WorkingDirectory = "${pkgs.funkwhale}/api";
+          };
+        in {
+        "funkwhale.target" = {
+          description = "Funkwhale";
+          wants = ["funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service"];
+        };
+        funkwhale-psql-init = {
+          description = "Funkwhale database preparation";
+          after = [ "redis.service" "postgresql.service" ];
+          wantedBy = [ "funkwhale-init.service" ];
+          before   = [ "funkwhale-init.service" ];
+          serviceConfig = {
+            User = "postgres";
+          };
+          script = ''
+            ${pkgs.postgresql}/bin/psql -d funkwhale -c 'CREATE EXTENSION IF NOT EXISTS "unaccent";'
+          '';
+        };
+        funkwhale-init = {
+          description = "Funkwhale initialization";
+          after = [ "funkwhale-psql-init.service" ];
+          wantedBy = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          before   = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          environment = funkwhaleEnv;
+          script = ''
+            chmod a+rx ${funkwhaleHome}
+            if ! test -e ${cfg.api.media_root}; then
+            mkdir -p ${cfg.api.media_root}
+            mkdir -p ${cfg.api.static_root}
+            mkdir -p ${cfg.music_directory_path}
```

Comment:
Look into `systemd.tmpfiles.rules` to create these directories for you.

Reply:
thanks. fixed.